### PR TITLE
Permit `urllib3.Timeout` instances for defining timeout values

### PR DESCRIPTION
**Pull request by amotl**
## About

Permit `urllib3.Timeout` instances for defining timeout values. Using that, you can configure both the socket `connect` and `read` timeout settings, in seconds.

## Problem
When defining the `timeout` value as `urllib3.Timeout` instance,
```python
HttpClient(crate_host, timeout=urllib3.Timeout(connect=0.01, read=None))
```
this error is raised:
```python
float() argument must be a string or a real number, not 'Timeout'
```

## References
- Needed by: https://github.com/crate/crash/pull/421

### CODE REVIEW

File: docs/by-example/http.rst [hunk 1]
```diff
@@ -199,8 +199,8 @@ timeout exception:
     {...}
     >>> http_client.close()
 
-It's possible to define a HTTP timeout in seconds on client instantiation, so
-an exception is raised when the timeout is reached:
+It is possible to define a HTTP timeout in seconds when creating a client
+object, so an exception is raised when the timeout expired:
```
Comment (matriv) [inline docs/by-example/http.rst:203]:
```suggestion
object, so an exception is raised when the timeout expires:
```

File: CHANGES.txt [hunk 2]
```diff
@@ -5,6 +5,9 @@ Changes for crate
 Unreleased
 ==========
 
+- Stopped decoding the ``timeout`` parameter strictly as scalar to also
```
Comment (matriv) [inline CHANGES.txt:8]: Please use a user facing message here, so what has changed for the user, not the internal implementation details.

**Review by surister — APPROVED** [review]

File: CHANGES.txt (hunk 2 above)
Comment (amotl) [reply CHANGES.txt]: Thanks. I've amended both the changelog item, and the commit message according to your suggestions.

File: CHANGES.txt [hunk 3]
```diff
@@ -5,6 +5,10 @@ Changes for crate
 Unreleased
 ==========
 
+- Permit ``urllib3.Timeout`` instances for defining timeout values.
+  Using that, you can configure both the socket ``connect`` and ``read``
```
Comment (matriv) [inline CHANGES.txt:9]: Maybe reads better:
```
This way, both ``connect`` and ``read`` socket timeout settings can be configured, in seconds.
```

**Review by matriv — APPROVED** [review]

File: CHANGES.txt (hunk 3 above)
Comment (amotl) [reply CHANGES.txt]: Thanks for your suggestion. I used it 1:1, yet separated the mention of the unit into a separate sentence.

> Permit ``urllib3.Timeout`` instances for defining timeout values.
> This way, both ``connect`` and ``read`` socket timeout settings can be
> configured. The unit is seconds.

**Review by mfussenegger — APPROVED** [review]

**Pull request merged**